astropy/io/fits/file.py

```diff
@@ -97,6 +104,9 @@ def __init__(self, fileobj=None, mode=None, memmap=None, clobber=False,
             return
         else:
             self.simulateonly = False
+            # If fileobj is of type pathlib.Path
+            if PY3 and HAS_PATHLIB and isinstance(fileobj, Path):
```

Comment:
You don't need to check `PY3`, all that matters is `HAS_PATHLIB` to make sure the second logical test will execute properly.